### PR TITLE
Cleanup build scan publishing logic

### PR DESCRIPTION
**Pull request by blindpirate**
After https://github.com/gradle/gradle-enterprise-conventions-plugin/pull/8
we can pass `-DpublishStrategy=publishOnFailure`. In this way we can avoid
extra logic in the build.

### CODE REVIEW

**Comment by blindpirate** [comment]
@bot-gradle test QF

File: .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt [hunk 1]
```diff
@@ -150,7 +150,7 @@ fun BaseGradleBuildType.killProcessStepIfNecessary(stepName: String, os: Os = Os
             gradleWrapper {
                 name = stepName
                 executionMode = BuildStep.ExecutionMode.ALWAYS
-                tasks = "killExistingProcessesStartedByGradle"
+                tasks = "killExistingProcessesStartedByGradle -DpublishStrategy=publishOnFailure" // https://github.com/gradle/gradle-enterprise-conventions-plugin/pull/8
```
Comment (wolfs) [inline .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt:153]: Do you know what the difference between `tasks` and `gradleParams` means? Given the name, I would expect the system property go to `gradleParams`.

Comment (blindpirate) [reply .teamcity/Gradle_Check/configurations/GradleBuildConfigurationDefaults.kt]: They have no differences when merged into the final parameters, but yes, it SHOULD be in `gradleParams`.

**Comment by blindpirate** [comment]
@bot-gradle test QFL

**Comment by blindpirate** [comment]
@bot-gradle test QF

**Comment by blindpirate** [comment]
Thanks @wolfs !

**Pull request merged**